ionoscloud/data_source_certificate_manager_auto_certificate.go

```diff
@@ -0,0 +1,121 @@
+				Type:             schema.TypeString,
+				Optional:         true,
+				Description:      "The ID of the auto-certificate",
+				ValidateDiagFunc: validation.ToDiagFunc(validation.IsUUID),
+			},
+			"name": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Description: "The name of the auto-certificate",
+			},
+			"common_name": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "The common name (DNS) of the certificate to issue. The common name is a part of a zone in IONOS Cloud DNS",
+			},
+			"key_algorithm": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "The key algorithm used to generate the certificate",
+			},
+			"subject_alternative_names": {
+				Type:     schema.TypeList,
+				Computed: true,
+				Elem: &schema.Schema{
+					Type: schema.TypeString,
+				},
+				Description: "Additional names added to the issued certificate. The additional names are part of a zone in IONOS Cloud DNS",
+			},
+			"last_issued_certificate_id": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "The ID of the last certificate that was issued",
+			},
+			"provider_id": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "The certificate provider used to issue the certificates",
+			},
+		},
+		Timeouts: &resourceDefaultTimeouts,
+	}
+}
+
+func dataSourceAutoCertificateRead(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	client := meta.(services.SdkBundle).CertManagerClient
+	location := d.Get("location").(string)
+	id, idOk := d.GetOk("id")
+	name, nameOk := d.GetOk("name")
+
+	if idOk && nameOk {
+		return diag.FromErr(fmt.Errorf("ID and name cannot be provided at the same time"))
+	}
+	if !idOk && !nameOk {
+		return diag.FromErr(fmt.Errorf("please provide either the auto-certificate ID or name"))
+	}
+
+	var autoCertificate certSDK.AutoCertificateRead
+	var err error
+
+	if idOk {
+		id := id.(string)
+		autoCertificate, _, err = client.GetAutoCertificate(ctx, id, location)
+		if err != nil {
+			return diag.FromErr(fmt.Errorf("an error occurred while fetching the auto-certificate with ID: %v, error: %w", id, err))
+		}
+	} else {
+		autoCertificates, _, err := client.ListAutoCertificates(ctx, location)
+		if err != nil {
+			return diag.FromErr(fmt.Errorf("an error occurred while fetching auto-certificates: %w", err))
+		}
+		var results []certSDK.AutoCertificateRead
+		if autoCertificates.Items != nil {
+			for _, autoCertificateItem := range *autoCertificates.Items {
+				if autoCertificateItem.Properties != nil && autoCertificateItem.Properties.Name != nil && strings.EqualFold(*autoCertificateItem.Properties.Name, name.(string)) {
+					results = append(results, autoCertificateItem)
+				}
+			}
+		}
+
+		if len(results) == 0 {
+			return diag.FromErr(fmt.Errorf("no auto-certificate found with the specified name: %v", name))
+		} else if len(results) > 1 {
```

Comment:
you can use just if without the else

Reply:
@digna-ionos please leave a suggestion with how the code should look like, it is not clear for me from your comment. I'm interested in the case in which the `len(results) is 1`, the error cases are `== 0` and `> 1`, not sure how can I do this check without an if/else, so please leave the exact code so I can understand properly. Thanks!

Reply:
`
if len(results) == 0 {
  return ...
}
if len(results) > 1 {
  return ...
}
`

Reply:
@digna-ionos thanks, I'll make the change.

Reply:
@digna-ionos done in ec747230649ad96067a92e1d3b11d469d3eda2f1